src/Components/Users/UsersForm.js

```diff
@@ -1,61 +1,169 @@
 import React, { useState } from 'react';
 import '../FormStyles.css';
+import { Field, Form, Formik } from 'formik';
+import { Alert } from '@mui/material';
+import { UserFormSchema } from './UserFormSchema';
+import { CustomDropzone } from './CustomDropzone';
+import axios from 'axios';
 
-const UserForm = () => {
-  const [initialValues, setInitialValues] = useState({
-    name: '',
-    email: '',
-    roleId: '',
+const UserForm = ({ user = null }) => {
+  const [success, setSuccess] = useState({
+    status: false,
+    errors: null,
   });
+  const userValues = user
+    ? { ...user, role: user.role_id }
+    : {
+        name: '',
+        email: '',
+        password: '',
+        role: '',
+        image: '',
+      };
 
-  const handleChange = (e) => {
-    if (e.target.name === 'name') {
-      setInitialValues({ ...initialValues, name: e.target.value });
-    }
-    if (e.target.name === 'email') {
-      setInitialValues({ ...initialValues, email: e.target.value });
+  const handleSubmit = async (values, formik) => {
+    const url = 'http://ongapi.alkemy.org/api/';
+
+    try {
+      let response;
+
+      if (user && user.id) {
+        response = await axios.patch(
+          `${url}users/${user.id}`,
+          JSON.stringify({ ...values, role_id: values.role }),
+          {
+            headers: {
+              'Content-Type': 'application/json',
+            },
+          },
+        );
+      } else {
+        response = await axios.post(`${url}users`, JSON.stringify(values), {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+      }
+
+      if (response.status === 200) {
+        setSuccess({
+          ...success,
+          status: true,
+        });
+        formik.resetForm();
+        clearImage();
+        setTimeout(() => {
+          setSuccess({
+            ...success,
+            status: false,
+          });
+        }, 4000);
+      } else {
+        return new Error('Error');
+      }
+    } catch (e) {
+      setSuccess({
+        status: false,
+        errors: true,
+      });
+      setTimeout(() => {
+        setSuccess({
+          status: false,
+          errors: null,
+        });
+      }, 4000);
     }
```

Comment:
this should be on userService. If it already exists, use the functions from there